sycl/include/CL/sycl/atomic_ref.hpp

```diff
@@ -0,0 +1,662 @@
+//==----- atomic_ref.hpp - SYCL 2020 atomic_ref ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/sycl/access/access.hpp>
+#include <CL/sycl/atomic.hpp>
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/detail/spirv.hpp>
+#include <CL/sycl/memory_enums.hpp>
+#ifdef __SYCL_DEVICE_ONLY__
+#include <CL/sycl/multi_ptr.hpp>
+#endif
```

Comment:
Is this some kind of an optimization? If so, should we move include of `CL/sycl/detail/spirv.hpp` into this condition as well?

Reply:
@AlexeySachkov Thank you for the idea, I've moved the import under the `__SYCL_DEVICE_ONLY__` guard.